src/parameterizations/lateral/MOM_MEKE.F90

```diff
@@ -573,7 +580,13 @@ subroutine MEKE_equilibrium(CS, MEKE, G, GV, US, SN_u, SN_v, drag_rate_visc, I_m
     !SN = 0.25*max( (SN_u(I,j) + SN_u(I-1,j)) + (SN_v(i,J) + SN_v(i,J-1)), 0.)
     ! This avoids extremes values in equilibrium solution due to bad values in SN_u, SN_v
     SN = min( min(SN_u(I,j) , SN_u(I-1,j)) , min(SN_v(i,J), SN_v(i,J-1)) )
-    beta = sqrt( G%dF_dx(i,j)**2 + G%dF_dy(i,j)**2 )
+
+    FatH = 0.25*((G%CoriolisBu(i,j) + G%CoriolisBu(i-1,j-1)) + &
+           (G%CoriolisBu(i-1,j) + G%CoriolisBu(i,j-1))) !< Coriolis parameter at h points
+    beta = sqrt( ( G%dF_dx(i,j) - CS%MEKE_topographic_beta*FatH/D(i,j)*(D(i+1,j) - D(i-1,j)) &
```

Comment:
We could probably precalculate this constant-in-time factor and store it in the MEKE type. And/or pre-calculate the f/D *dD_dx contribution and store that in G. The two-point difference has a null-space but it's not obvious what a good alternative is here.

Reply:
Thank you Alistair @adcroft for pointing out the mistake and the suggestions! I've removed the use of D, lat, and lon so the memory is saved the most. Shall I submit a new pull request?

> The pre-calculation of c_beta

For now we don't think it's a straightforward thing to do - there can be a full theory about this scale factor and Malte told me Andrew Stewart and a group of other people are working on refining the MEKE recently and maybe they have some thoughts on it? And that's what motivated us to submit the pull request here, so that people interested can help improve it. 

>The pre-calculation of f/D * dD_dx

Do you mean to compute it here in this module & in the relevant MEKE subroutines? Yet I don't think it may increase the computational efficiency much since the computation of beta* is performed in two for-loops and so does that of f/D * dD_dx. Unless it will be used elsewhere, separating the calculation of f/D * dD_dx will only involve more computational hours?

>2-point differential

I agree that no ideal alternative to the 2 point differential is easily found...

Reply:
I'll also include Malte in this thread... @MFJansen

Reply:
No need for new PR; pushing updates (as you have already) to your branch is automatically revising the PR. You should, however, search and remove trailing spaces which is causing the red "x" to appear on your commits. See the red text in the log at https://travis-ci.org/NOAA-GFDL/MOM6/builds/454730193?utm_source=github_status&utm_medium=notification